src/sign.jl

```diff
@@ -121,13 +121,13 @@ function sign_aws4!(r::AWSRequest, t)
     signed_headers = join(sort([lowercase(k) for k in keys(r[:headers])]), ";")
 
     # Sort Query String...
-    uri = URI(r[:url])
+    uri = URI(replace(r[:url], " ", "+"))
     query = query_params(uri)
     query = [(k, query[k]) for k in sort(collect(keys(query)))]
 
     # Create hash of canonical request...
     canonical_form = string(r[:verb], "\n",
-                            escape_with(uri.path, path_esc_chars), "\n",
+                            escape_with(replace(uri.path, "+", " "), path_esc_chars), "\n",
```

Comment:
This is slightly weird, in that we are unescaping what we previously escaped. However, we needed to escape earlier since we are creating an URI object, which fails if the path is not escaped. But when creating the canonical form, we want to start with the raw values, so that the canonical form is consistent.